### DISCUSSION TITLE
How to modify the default instance

### DISCUSSION THREAD

**Discussion by tekjock** [paukstelis/octoprint_deploy#55] [General]
Just getting started with octoprint_deploy, well, just getting started with octoprint on ubuntu,

I have the base instance and I wanted to install  some plugins before starting to add new instances.

**Comment by paukstelis** [comment]
just connect to http://ipaddress:5000 to modify the template instance.

**Comment by tekjock** [comment]
I tried that, the plugin manger shows one I installed (Orange box), but it does not show as installed on the lower left. 

ALSO,, durring the setup, I chose UI Customizer, uninstalled it, but still shows on the lower left (black arrrow)

![image](https://user-images.githubusercontent.com/49789474/216790148-3a4f0347-6e47-40db-bada-db2595f4e641.png)

**Comment by tekjock** [comment]
OK< i guess I had to restart the server .. which kind of stinks when I install a plugin when I want to run hopfully up to 3 printers

**Reply by paukstelis** [reply]
I don't think you quite understand how it works. Any plugin you install on any instance will be available to all instances. Any plugin SETTINGS you modify in the template instance will go to ALL new instances you make. Plugin settings modifications made in a non-template instance stay with that instance. The ONLY reason to install plugins on the template is if you have some specific settings that you want all your instances to have. The template instance is also the only one that has to be restarted manually, though you should really never have to start it again after  you have made new instances.

**Reply by tekjock** [reply]
I do get how Octoprint works, but, this is my first time using it on Ubuntu. i have been using Octoprint for around a year, 1 on RaspPi and one on Windows. I have not even added any instances yet, I was just trying to get things setup before hand.

**Reply by paukstelis** [reply]
I was not saying you didn't understand OctoPrint, just how plugins work with a multi-instance OctoPrint using a shared venv. This talk I gave at ERRF would be a good resource: https://www.youtube.com/watch?v=q0iCNl8-kJI&t=15378s

**Reply by tekjock** [reply]
Thanks will check it out. 

but all is well now, I have 2 printers now running on my Ubuntu server ... I hvae a 3rd printer, but that is on a RaspPi w/ Pi Cam, going to print with the 2 on there before I add a 3rd

**Comment by paukstelis** [comment]
You have to restart the instance after installing or removing plugins. The template cannot be restarted from the UI.
`sudo systemctl restart octoprint_default`

**Reply by tekjock** [reply]
its all good now, just learning how things work on a ubuntu server. Not that great with Linux
